.github/workflows/pkg-ci.yml

```diff
@@ -3,11 +3,25 @@ on:
   pull_request:
 jobs:
   build:
-    runs-on:
-      - ubuntu-latest
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        # runner architecture defined here -> https://github.com/actions/runner-images
+        os: [
+          ubuntu-22.04,     # linux-amd64
+          ubuntu-22.04-arm, # linux-arm64
+          macos-13,         # darwin-amd64
+          macos-13-xlarge   # darwin-arm64
+        ]
```

Comment:
I'm going to leave this matrix on the ci job to make the main `pkg` pipeline as it actually caught something that having the single runner wouldn't of.